### PR TITLE
Feat/add orchestrator unit tests

### LINKED ISSUE #38: Add unit tests for `DownloadOrchestrator`
The `DownloadOrchestrator` (`forklet.core.orchestrator.DownloadOrchestrator`) is a critical component that manages the entire download process with concurrency, error handling, and progress tracking. Currently, it does not have direct unit test coverage.

To improve reliability and ensure future changes are safe, we should add **basic but meaningful unit tests** for this orchestrator.

### ✅ Proposed Tests

* **Initialization**

  * Verify that `max_concurrent_downloads` is correctly set.
  * Ensure semaphore is initialized properly.

* **`execute_download`**

  * Simulate a download with mocked `GitHubAPIService` and `DownloadService`.
  * Ensure filtered files are passed to `_download_files_concurrently`.
  * Verify that a `DownloadResult` with correct status is returned.
  * Test error handling when repository tree fetch fails.

* **`_download_files_concurrently`**

  * Verify successful downloads increment stats and progress.
  * Ensure failed downloads are recorded properly in `failed_files`.
  * Test cancellation (`self._is_cancelled = True`) stops processing.

* **`_download_single_file`**

  * Test skip logic when file already exists and `overwrite_existing=False`.
  * Ensure file content is written via `DownloadService.save_content`.
  * Verify progress and stats are updated after a successful download.
  * Test error case when `get_file_content` raises.

* **Control Methods**

  * `cancel()` → sets `_is_cancelled=True` and logs.
  * `pause()` and `resume()` → check that they run without errors.
  * `get_current_progress()` → returns `None` (until implemented).

### 🎯 Benefits

* Increases confidence in Forklet’s core logic.
* Ensures concurrency and error handling behave as expected.
* Makes it easier for contributors to safely modify orchestrator logic.

### PR DESCRIPTION
**Pull request by ArjunDixit1**
Closes #38 

This PR introduces a comprehensive suite of **unit tests for the DownloadOrchestrator class**, addressing the lack of test coverage for this critical component. The new tests verify its core functionality—including concurrency, error handling, and control methods—which will improve code reliability and make future changes safer to implement.

### Key Changes
The new test suite (tests/core/test_download_orchestrator.py) covers the following scenarios:

**Initialization**: Verifies that the orchestrator is set up correctly with the specified concurrency limit.

**execute_download Logic:**

Tests the main success path ("happy path").

Handles failure scenarios like repository fetch errors.

Covers edge cases such as empty repositories and partial download failures.

**_download_single_file Logic:**

Tests the file skipping mechanism (overwrite_existing=False).

Verifies content saving and error handling during individual file downloads.

**Control Methods**: Ensures the cancel(), pause(), and resume() methods are functional.

### How to Verify
Pull down this branch.

Install the development dependencies: **pip install -e .[dev]**

Run the full test suite and confirm that all tests pass:
image

### CODE REVIEW

**Comment by Einswilli** [comment]
Seems tests failed for `tests/core/test_download_orchestrator.py` (is your repo up to date please???)

File: tests/infrastructure/test_rate_limiter.py [hunk 1]
```diff
@@ -108,51 +101,53 @@ async def test_update_rate_limit_increments_consecutive_limits():
 # ------------------------------------------
 @pytest.mark.asyncio
 @patch('asyncio.sleep', new_callable=AsyncMock)
+@pytest.mark.asyncio
```
Comment (Einswilli) [inline tests/infrastructure/test_rate_limiter.py:104]: I think this is unnecessary

File: tests/infrastructure/test_rate_limiter.py
Comment (Einswilli) [inline tests/infrastructure/test_rate_limiter.py:1]: I think this file is not a part of the PR

**Review by Einswilli — CHANGES_REQUESTED** [review]

File: tests/infrastructure/test_rate_limiter.py (hunk 1 above)
Comment (ArjunDixit1) [reply tests/infrastructure/test_rate_limiter.py:104]: I have changed one statement in for my orchestrator test file to work so its not unnecessary  @Einswilli

File: tests/infrastructure/test_rate_limiter.py
Comment (ArjunDixit1) [reply tests/infrastructure/test_rate_limiter.py:1]: You're right, this wasn't part of the original scope. However, while working on the orchestrator tests, I discovered that the tests in test_rate_limiter.py were failing in the workflow.
I've included the fixes in this PR to ensure the entire test suite is stable and passes correctly.

**Review by Einswilli — APPROVED** [review]
Thanks for your Contributions boss
